# Challenge 1: Files, Loops and Conditional Logic
For this challenge, you'll be working with the data set behind this FiveThirtyEight article on common unisex (gender-neutral) names in the United States. You'll start by reading in the file and iteratively converting the data to more useful representations. At the end of this challenge, you'll filter the data so that it only includes the names that at least 1,000 people share.

The staff at FiveThirtyEight compiled this data set from information at the Social Security Adminstration's website. You'll work with a shortened version of the full data set to complete this challenge.

Dataset is available at https://github.com/fivethirtyeight/data/blob/master/unisex-names/unisex_names_table.csv 
but is downloaded into the local directory

This version of the files has all the fields since the file is downloaded from github




In [1]:
'''
Make sure there are no empty rows at the end of the file because the split function will throw an error. 
One way to make sure it doesn't happen is to open it in excel and save it as csv again
'''
# Open the file
file_unisex_names_table_csv = open("dq_unisex_names.csv", "r")

# Read the File
file_data = file_unisex_names_table_csv.read()

# Split the data 
rows_in_file = file_data.split("\n")


#Print the first 5 rows
print(rows_in_file[0:5])

[',name,total,male_share,female_share,gap', '1,Casey,176544.3281,0.584286566,0.415713434,0.168573132', '2,Riley,154860.6652,0.507639071,0.492360929,0.015278142', '3,Jessie,136381.8307,0.477834268,0.522165732,0.044331463', '4,Jackie,132928.7887,0.421132602,0.578867398,0.157734796']


In [2]:
# Split the data in each row to create a list of lists

final_list = []

for item in rows_in_file:
    split_list = item.split(",")
    final_list.append(split_list)
    
#Print the last 5 rows of the final list
print(final_list[len(final_list)-5:len(final_list)])


[['915', 'Eaden', '102.319579', '0.572760615', '0.427239385', '0.14552123'], ['916', 'Inioluwa', '101.384201', '0.352728923', '0.647271077', '0.294542155'], ['917', 'Gwin', '101.243364', '0.562136576', '0.437863424', '0.124273152'], ['918', 'Yacine', '100.2304', '0.544598944', '0.455401056', '0.089197888'], ['919', 'Aeon', '100.21104', '0.464834563', '0.535165437', '0.070330874']]


In [3]:
'''
Since the first row are labels remove them so the float conversions are easy otherwise an error is thrown 
when you convert it to float
'''

# Create a new list which contains only the name and the total

final_list_with_no_headings = final_list[1:len(final_list)]
print("Final list with no headings list size : " + str(len(final_list)))

short_list = []
for item in final_list_with_no_headings:
    name = item[1]
    total = float(item[2])
    new_list = [name,total]
    short_list.append(new_list)
    
print(short_list[0:5])

Final list with no headings list size : 920
[['Casey', 176544.3281], ['Riley', 154860.6652], ['Jessie', 136381.8307], ['Jackie', 132928.7887], ['Avery', 121797.4195]]


In [5]:
# Create the final list with only the names that have occurred more than 1000
thousand_or_greater = []
for item in short_list:
    if item[1] >= 1000:
        thousand_or_greater.append(item)
        
print(thousand_or_greater[0:5])

print("Thousand or greater list size : " + str(len(thousand_or_greater)))
print("Total list :" + str(len(final_list)))
print("Percentage of people whose names occur more than 1000 : " + str((len(thousand_or_greater)/(len(final_list)) * 100)))

[['Casey', 176544.3281], ['Riley', 154860.6652], ['Jessie', 136381.8307], ['Jackie', 132928.7887], ['Avery', 121797.4195]]
Thousand or greater list size : 291
Total list :920
Percentage of people whose names occur more than 1000 : 31.630434782608695


# Findings:
Total List size is 920
Thousand or greater name list is 291
Percentage of people whose name has occured more than 1000 is 31.63%